packages/next/build/webpack/loaders/next-image-loader.js

```diff
@@ -0,0 +1,58 @@
+import loaderUtils from 'next/dist/compiled/loader-utils'
+import sizeOf from 'image-size'
+import { processBuffer } from '../../../next-server/server/lib/squoosh/main'
+import fs from 'fs'
+
+async function nextImageLoader(content) {
+  this.cacheable && this.cacheable(true)
+  this.addDependency(this.resourcePath)
+
+  const context = this.rootContext
+  const opts = { context, content }
+  const interpolatedName = loaderUtils.interpolateName(
+    this,
+    '/[path][name].[ext]',
+    opts
+  )
+
+  let extension = loaderUtils.interpolateName(this, '[ext]', opts)
+  if (extension === 'jpg') {
+    extension = 'jpeg'
+  }
+
+  if (interpolatedName.slice(0, 7) !== '/public') {
+    const err = new Error(
+      'Static Image loader used with filepath not in the /public directory: ' +
+        interpolatedName
+    )
+    this.emitError(err)
+    return
+  }
+
+  const src = interpolatedName.slice(7)
+  const imageSize = sizeOf(this.resourcePath)
+  let dataURI
+  if (extension === 'jpeg' || extension === 'png') {
+    const fileBuffer = Buffer.from(fs.readFileSync(this.resourcePath))
```

Comment:
You should avoid sync fs calls for performance reasons. You don't need to read this file anyway, as you get it as argument to the loader -> `content`
You can switch the loader to `raw` mode by setting `raw = true` property on the loader or as export, to get the file as Buffer instead of a string.

Reply:
I tried enabling raw mode with both of the following:
```
nextImageLoader.raw = true
export default nextImageLoader
```
and
```
module.exports = nextImageLoader
module.exports.raw = true
```
and in both cases it didn't seem to work right. I was still getting `typeof content` equal to `string`, and when I attempted to convert the string to a Buffer something's off. When I call `this.emitFile(interpolatedName, content, null)`, for instance, I get broken image files.
I've been looking at this for a bit, but I'm getting kind of stumped--do you have advice for troubleshooting here?